packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -269,44 +269,52 @@ export function HierarchicalTermSelector( { slug } ) {
 		setShowForm( ! showForm );
 	};
 
-	const onAddTerm = async ( event ) => {
-		event.preventDefault();
-		if ( formName === '' || adding ) {
-			return;
-		}
+	const { createErrorNotice } = wp.data.dispatch('core/notices'); // Importing dispatch function
 
-		// Check if the term we are adding already exists.
-		const existingTerm = findTerm( availableTerms, formParent, formName );
-		if ( existingTerm ) {
-			// If the term we are adding exists but is not selected select it.
-			if ( ! terms.some( ( term ) => term === existingTerm.id ) ) {
-				onUpdateTerms( [ ...terms, existingTerm.id ] );
+	const onAddTerm = async ( event ) => {
+		try {
```

Comment:
No need to wrap the whole method logic in `try/catch`, just the `addTerm`. You also need to pass an extra argument to the `addTerm`, so it throws an error.



Example:

```js
const addTerm = ( term ) => {
	return saveEntityRecord( 'taxonomy', slug, term, {
		throwOnError: true,
	} );
};

let newTerm;
try {
	newTerm = await addTerm( {
		name: formName,
		parent: formParent ? formParent : undefined,
	} );
} catch ( error ) {
	createErrorNotice( error.message, {
		type: 'snackbar',
	} );
	return;
}
```